erpnext/schools/doctype/assessment_result_tool/assessment_result_tool.js

```diff
@@ -7,17 +7,19 @@ cur_frm.add_fetch("assessment_plan", "student_batch", "student_batch");
 frappe.ui.form.on('Assessment Result Tool', {
 
 	onload: function(frm) {
+		frm.trigger("refresh");
+	},
+
+	refresh: function(frm) {
 		if (frappe.route_options) {
 			frm.doc.assessment_plan = frappe.route_options.assessment_plan;
+			frm.trigger("assessment_plan");
```

Comment:
it's not a good solution, it will reset the table values, if you just returning back from any other tabs or document without any changes.

may be you can use earlier `route_options` and on `refresh` check if routing from any other student group than selected, then trigger the "assessment_plan". And always clear the route options.